integration-tests/common/src/main/kotlin/me/tatarka/inject/test/Scope.kt

```diff
@@ -236,3 +236,25 @@ abstract class SuperclassScopeFooComponent : BaseSuperclassScopeFooComponent {
 
     companion object
 }
+
+@Scope
+annotation class ScopeWithArg(val value: String)
+
+@Inject
+@ScopeWithArg("parent")
+class ScopedParentFoo
+
+@Inject
+@ScopeWithArg("child")
+class ScopedChildFoo
+
+@Component
+@ScopeWithArg("parent")
```

Comment:
Curious, `KClass` arguments are supported as well, right? I didn't see how you implement the `equals()` check.

Reply:
They should be yes, it's using the same logic that was already used for qualifiers

Reply:
If you are curious https://github.com/evant/kotlin-inject/blob/094d8ce2c1a70a1b29fc08be073052fcf3a76e4a/ast/ksp/src/main/kotlin/me/tatarka/kotlin/ast/KSUtil.kt#L60

Reply:
Thanks!